update.ps1

```diff
@@ -1,201 +1,532 @@
+    if (($correlatedAccount | Measure-Object).count -eq 1) {
+        $actionMessage = "comparing current account to mapped properties"
+
+        # Set Previous data (if there are no changes between PreviousData and Data, HelloID will log "update finished with no changes")
+        $outputContext.PreviousData = $correlatedAccount.PsObject.Copy()
+
+        if ($actionContext.Configuration.updateManagerOnUpdate -eq $true) {
+            #region Get manager of account
+            # API docs: https://mwpapi.kpnwerkplek.com/index.html, specific API call: GET /api/users/{identifier}/manager
+            $actionMessage = "querying manager of account with AccountReference: $($actionContext.References.Account | ConvertTo-Json)"
+    
+            $getKPNLisaAccountManagerSplatParams = @{
+                Uri         = "$($actionContext.Configuration.MWPApiBaseUrl)/users/$($actionContext.References.Account)/manager"
+                Method      = "GET"
+                Verbose     = $false
+                ErrorAction = "Stop"
+            }
+    
+            Write-Verbose "SplatParams: $($getKPNLisaAccountSplatParams | ConvertTo-Json)"
+    
+            # Add header after printing splat
+            $getKPNLisaAccountManagerSplatParams['Headers'] = $headers
+    
+            $getKPNLisaAccountManagerResponse = $null
+            $getKPNLisaAccountManagerResponse = Invoke-RestMethod @getKPNLisaAccountManagerSplatParams
+            $previousManagerId = $getKPNLisaAccountManagerResponse.id
+            
+            Write-Verbose "Queried manager of account with AccountReference: $($actionContext.References.Account | ConvertTo-Json). Result: $($getKPNLisaAccountManagerResponse | ConvertTo-Json)"
+            #endregion Get manager of  account
+    
+            #region Calulate manager action
+            if ($actionContext.References.ManagerAccount -ne $previousManagerId) {
+                if (-not[String]::IsNullOrEmpty(($actionContext.References.ManagerAccount))) {
+                    $actionManager = "Update"
+                }
+                else {
+                    $actionManager = "Clear"
+                }
+            }
+            else {
+                $actionManager = "NoChanges"
+            }
+            #endregion Calulate manager action
 
-    Write-Verbose -Verbose "Updating KPN Lisa account for '$($personContext.Person.DisplayName)'"
+            # Add managerId to Previous data
+            $outputContext.PreviousData | Add-Member -MemberType NoteProperty -Name "managerId" -Value $previousManagerId -Force
+        }
 
-    $SplatParams = @{
-        Uri    = "$($actionContext.Configuration.BaseUrl)/Users/$($actionContext.References.Account)/bulk"
-        Method = "Patch"
-        Body   = $outputContext.Data | Select-Object -Property * -ExcludeProperty $NonUpdatables
-    }
+        # Create reference object from correlated account
+        $accountReferenceObject = [PSCustomObject]@{}
```

Comment:
Voor het accountReferenceObject en accountDifferenceObject, dat kunnen we misschien efficiënter doen omdat ze eigenlijk niet heel anders zijn dan correlatedAccount en account.

Reply:
Efficiënter vast en zeker. Maar correlatedAccount en Account zijn niet 100% gelijk in de opbouw, bijv. de array van mailadressen, dit forceren we door zo het object op te bouwen.
Aangezien het met dit getest is en werkt, laat ik dit verder zo.